src/overlay/FlowControl.h

```diff
@@ -54,7 +72,12 @@ class FlowControl
     std::shared_ptr<FlowControlCapacity> mFlowControlCapacity;
     std::shared_ptr<FlowControlByteCapacity> mFlowControlBytesCapacity;
 
-    Application& mApp;
+    OverlayMetrics& mOverlayMetrics;
+    OverlayManager& mOverlayManager;
+    Config const mConfig;
```

Comment:
In general, in a multi-threaded environment, I prefer a copy to a reference to avoid auditing the code for thread-safety and prevent footguns. This works when performance isn't an issue of course: in this case I think it's fine since Config is a tiny const object stored in each peer.

I'd ideally like to do the same for OverlayMetrics (e.g. deal with copies and do message passing of sorts), but that's more involved, so maybe as a follow-up.